frontend/e2e/exercises-page.ts

```diff
@@ -22,4 +22,25 @@ export class ExercisesPage {
     async checkExercisePageIsVisible(exercise: string): Promise<void> {
         await expect(this.page.locator(`mat-toolbar .title:has-text("${exercise}")`).first()).toBeVisible();
     }
+
+    async openFilterBar(): Promise<void> {
+        await this.page.click('mat-expansion-panel-header .mat-expansion-indicator');
+    }
+
+    async updateQuery(query: string): Promise<void> {
+        await this.page.fill('.example-form-field .query-input', query);
```

Comment:
**suggestion:** would be good to always point to the actual input element over using a class to make sure the right element gets the event. E.g.

```
await this.page.fill('.example-form-field .query-input input', query);
```
or
```
await this.page.fill('.example-form-field input.query', query);
```
depending on which one is right here.